### DISCUSSION TITLE
Badges' appropriate href

### DISCUSSION THREAD

**Discussion by jihoon-seo** [cloud-barista/cb-tumblebug#389] [General]
You can see that hyperlinks are enabled for some badges, but not for others.
(뱃지를 누르면 정보 페이지로 넘어가는 경우도 있지만, 뱃지 이미지를 새 탭으로 띄워주는 경우도 있음)

https://raw.githubusercontent.com/cloud-barista/cb-tumblebug/master/README.md

[![Go Report Card](https://goreportcard.com/badge/github.com/cloud-barista/cb-tumblebug)](https://goreportcard.com/report/github.com/cloud-barista/cb-tumblebug)
![Top Language](https://img.shields.io/github/languages/top/cloud-barista/cb-tumblebug) 
![Repo Size](https://img.shields.io/github/repo-size/cloud-barista/cb-tumblebug)
![GitHub go.mod Go version](https://img.shields.io/github/go-mod/go-version/cloud-barista/cb-tumblebug?label=go.mod)
[![GoDoc](https://godoc.org/github.com/cloud-barista/cb-tumblebug?status.svg)](https://pkg.go.dev/github.com/cloud-barista/cb-tumblebug)

![Release Version](https://img.shields.io/github/v/release/cloud-barista/cb-tumblebug)
[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://github.com/cloud-barista/cb-tumblebug/blob/master/LICENSE)

```Markdown
[![Go Report Card](https://goreportcard.com/badge/github.com/cloud-barista/cb-tumblebug)](https://goreportcard.com/report/github.com/cloud-barista/cb-tumblebug)
![Top Language](https://img.shields.io/github/languages/top/cloud-barista/cb-tumblebug) 
![Repo Size](https://img.shields.io/github/repo-size/cloud-barista/cb-tumblebug)
![GitHub go.mod Go version](https://img.shields.io/github/go-mod/go-version/cloud-barista/cb-tumblebug?label=go.mod)
[![GoDoc](https://godoc.org/github.com/cloud-barista/cb-tumblebug?status.svg)](https://pkg.go.dev/github.com/cloud-barista/cb-tumblebug)

![Release Version](https://img.shields.io/github/v/release/cloud-barista/cb-tumblebug)
[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://github.com/cloud-barista/cb-tumblebug/blob/master/LICENSE)
```

---

My suggestions are:

- Top Language: https://github.com/cloud-barista/cb-tumblebug/search?l=go
- Repo Size: Hmm..
  - https://stackoverflow.com/questions/8646517/how-can-i-see-the-size-of-a-github-repository-before-cloning-it
  - https://api.github.com/repos/cloud-barista/cb-tumblebug
  - 뱃지 이미지를 새 탭으로 띄워주는 현 상태 대신, 아무 곳으로도 가지 않는 빈 링크로 바꾸는 방법도 있습니다.
- GitHub go.mod Go version: https://github.com/cloud-barista/cb-tumblebug/blob/master/go.mod
- Release Version: https://github.com/cloud-barista/cb-tumblebug/releases

---

Any comments are welcome!

**Comment by jihoon-seo** [comment]
- cb-operator 에 적용해 보았습니다.

https://github.com/cloud-barista/cb-operator/blob/master/README.md

- Top Language: cb-operator 는 HTML 로 표시되는데,
이는 `docker-compose-mode-files/data/grafana/data/plugins/*` 아래에 HTML 파일들이 많기 때문이라
cb-operator 의 실질적 top language 라고 하기에는 애매한 측면이 있어서
Top Language badge 를 삭제했습니다.
- Repo Size / Code Size: 뱃지 이미지를 새 탭으로 띄워주던 상태 대신, 아무 곳으로도 가지 않는 빈 링크로 바꿨습니다.
- [issue/open, issue/closed, pr/open, pr/closed] badge들을 추가하였습니다. (CB-Ladybug 참고)

**Reply by powerkimhub** [reply]
@jihoon-seo 

-일시적인 현상인지 모르겠지만, 현재 badge 링크중 not found가 있습니다. 
  - https://pkg.go.dev/github.com/cloud-barista/cb-operator

**Reply by seokho-son** [reply]
@powerkimhub 처음 사용하면 Fetch 가 필요해서 그렇습니다. 지금은 정상 동작할거예요

**Reply by jihoon-seo** [reply]
누군가가 해당 페이지에서 `Request "github.com/..."` 버튼을 한 번 눌러야 하는 것이었습니다.
제가 방금 눌렀습니다. :)

![image](https://user-images.githubusercontent.com/46767780/106981796-0ae5da80-67a6-11eb-90eb-7b360e008c03.png)

**Comment by seokho-son** [comment]
@jihoon-seo 
제안하신 방법대로 TB 리포에 적용하는 것이 좋아보입니다.. ^^

- Top Language: 
  https://github.com/cloud-barista/cb-tumblebug/search?l=go
- Repo Size:  
  뱃지 이미지를 새 탭으로 띄워주는 현 상태 대신, 아무 곳으로도 가지 않는 빈 링크로 바꾸는 방법
- GitHub go.mod Go version: 
  https://github.com/cloud-barista/cb-tumblebug/blob/master/go.mod
- Release Version: 
  https://github.com/cloud-barista/cb-tumblebug/releases

**Comment by jihoon-seo** [comment]
GoDoc badge 의 href 에 대해서는 다음의 3가지 선택지가 있습니다.
1. 마지막으로 찍은 tag (예: `v0.3.0-espresso`) 의 코드 형상 (https://github.com/cloud-barista/cb-tumblebug/commit/35e14a7c2542430e15b22cb56cd536048594fe1b) 에 따른 godoc
=> https://pkg.go.dev/github.com/cloud-barista/cb-tumblebug
2. `master` 브랜치의 **최신** 코드 형상에 따른 godoc
=> https://pkg.go.dev/github.com/cloud-barista/cb-tumblebug@master
3. 특정 tag 의 코드 형상에 따른 godoc => README.md badge href 를 업데이트해야 하는 번거로움 / 업데이트를 까먹을 가능성도 있음
=> https://pkg.go.dev/github.com/cloud-barista/cb-tumblebug@v0.3.0-espresso

**Reply by seokho-son** [reply]
v1.0.0 이전까지는 master로 제공하는 것이 어떨까요?
https://pkg.go.dev/github.com/cloud-barista/cb-tumblebug@master 

현재는 godoc 공개의 목적은 cb-tumblebug의 go pkg의 외부 공개(및 재활용)를 위해서라기 보다는, cb-tumblebug의 기여자들이 개발에 참고하기 위한 문서의 성향이 강하기 때문입니다.

master 브랜치의 페이지로 지정하여 사용하여도, 
godoc 웹사이트에서, 다른 Tag 버전으로 이동도 쉽게 가능하니.
Tag 또는 릴리스 브랜치 마다 따로 링크를 제공할 필요는 없지 않을까 생각됩니다.
(master 이외의 다른 브랜치의 경우에도, master로의 문서로 링크)

어떻게 생각하시나요?

**Reply by jihoon-seo** [reply]
좋아 보입니다!

---

이와 비슷한 것으로 Swagger doc 도 있습니다.

지금 확인해 보니,
- https://cloud-barista.github.io/download/ 페이지에서
Cloud-Barista v0.3.0-espresso 아래의 [API 규격] 을 누르면
- https://github.com/cloud-barista/docs/blob/master/technical_docs/API/CB-User_REST-API.md#v030-espresso 로 이동하며
여기서 [Namespace API] 를 누르면
- https://cloud-barista.github.io/cb-tumblebug-api-web/?url=https://raw.githubusercontent.com/cloud-barista/cb-tumblebug/master/src/docs/tbapi-0.3.0.yaml#/Namespace 로 이동합니다.
이 주소는 master 브랜치의 latest 한 tbapi-0.3.0.yaml 의 내용대로 Swagger doc 을 표시합니다.

~따라서, Swagger doc 도 현재는 master 로 제공되고 있다고 하겠습니다.~

그런데 tbapi-0.3.0.yaml 파일은
v0.3.0 릴리즈 시점에
[swagger.yaml](https://github.com/cloud-barista/cb-tumblebug/blob/master/src/docs/swagger.yaml) 파일을 수동으로 복사하여 생성한 파일입니다.

v0.3.0 릴리즈 이후로는
[swagger.yaml](https://github.com/cloud-barista/cb-tumblebug/blob/master/src/docs/swagger.yaml) 파일만 업데이트 되고 있습니다.

[정리]
- v0.3.0 에 대한 Swagger API Doc: https://cloud-barista.github.io/cb-tumblebug-api-web/?url=https://raw.githubusercontent.com/cloud-barista/cb-tumblebug/master/src/docs/tbapi-0.3.0.yaml
- 현재 master 브랜치에 대한 Swagger API Doc: https://cloud-barista.github.io/cb-tumblebug-api-web/?url=https://raw.githubusercontent.com/cloud-barista/cb-tumblebug/master/src/docs/swagger.yaml

**Reply by seokho-son** [reply]
잘 파악해주신 것 같습니다.

현재 master branch의 경우, master의 최신 상태 (swagger 문서를 잘 생성한 경우) 로 [API를 가이드](https://github.com/cloud-barista/cb-tumblebug#cb-tumblebug-%EB%B0%8F-cb-spider%EC%9D%98-rest-api%EB%A5%BC-%EC%82%AC%EC%9A%A9%ED%95%98%EC%97%AC-%ED%85%8C%EC%8A%A4%ED%8A%B8)하고 있습니다.

**Comment by jihoon-seo** [comment]
Swagger doc 뱃지도 추가하면 좋겠네요!

[예시]
`https://img.shields.io/badge/Swagger-API%20Doc-brightgreen`
[![Swagger API Doc](https://img.shields.io/badge/Swagger-API%20Doc-brightgreen)](https://cloud-barista.github.io/cb-tumblebug-api-web/?url=https://raw.githubusercontent.com/cloud-barista/cb-tumblebug/master/src/docs/swagger.yaml)

**Reply by seokho-son** [reply]
넵 ^^ CB-TB의 경우에는 적용 가능하겠네요. (API 문서의 정리가 좀 필요하지만..^^)
